### PR TITLE
[Merged by Bors] - Resolve issue_2168 with regression test

### PR DESCRIPTION
**Pull request by rrrodzilla**
- add: dev dependencies predicates, assert_cmd, rand
- fix: issue_2168

### CODE REVIEW

**Comment by sehz** [comment]
@tjtelan Can you review this PR?

**Review by tjtelan — CHANGES_REQUESTED** [review]
I needed to change one line in `fluvio-channel-cli`'s `Cargo.toml` to get the test to compile. Could you make that change please?

```diff
- fluvio-cli-common = { path = "../fluvio-cli-common", default-features = false }
+ fluvio-cli-common = { path = "../fluvio-cli-common"}
```

Also, thanks for writing a test! Could you add this test to [this makefile, in the `run-all-unit-test` target](https://github.com/rrrodzilla/fluvio/blob/7137d22309b50e998e5b2a153f6fccf8911818b3/makefiles/check.mk#L28) so we can run it during CI?

```diff
+ cargo test -p fluvio-channel-cli $(BUILD_FLAGS)
```

Otherwise I manually tested and it fixes issue #2168 ✅

LGTM!

**Comment by rrrodzilla** [comment]
Hey thanks for having a review.  I've made the requested changes with this last commit.  Thanks again!

File: crates/fluvio-channel-cli/tests/issue_2168_regression.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    //verify our test channel name is not currently in the channel config
+    let mut cmd = Command::cargo_bin("fluvio-channel")?;
+    cmd.arg("version").arg("list");
+    cmd.assert()
+        .success()
+        .stdout(predicate::str::contains(&random_channel_name).count(0));
+
+    //if we get this far, we can attempt to create the channel version
+    let mut cmd = Command::cargo_bin("fluvio-channel")?;
```
Comment (sehz) [inline crates/fluvio-channel-cli/tests/issue_2168_regression.rs:48]: same here

**Review by sehz — CHANGES_REQUESTED** [review]

**Comment by rrrodzilla** [comment]
Updates to the test with the requested changes pushed!

**Review by rrrodzilla — COMMENTED** [review]
I've updated the test to handle errors internally with `expect`

**Review by sehz — APPROVED** [review]

**Comment by sehz** [comment]
There is test failure: https://github.com/infinyon/fluvio/runs/5133172297?check_suite_focus=true

**Comment by rrrodzilla** [comment]
@tjtelan The test running from the makefile is expecting a channel config file in the CI environment and it's not finding one.  Is there not one or should I expect it in a different location?

**Comment by tjtelan** [comment]
> @tjtelan The test running from the makefile is expecting a channel config file in the CI environment and it's not finding one. Is there not one or should I expect it in a different location?

Yeah, the CI environment wouldn't have a channel file because we are only using the standard github runners. You'll need to create the config file when in the CI environment before running the test.

You can do a CI check by looking for the [`CI` environment variable](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables), which Github Actions will always set to `true`.

If that env var exists and it is set to `true`, you can run `fluvio-channel version switch stable`. I just tested locally, and it will create the `~/.fluvio` directory and the `channel` config for the rest of the test to use.

**Review by tjtelan — APPROVED** [review]

**Comment by sehz** [comment]
Good job.  Thanks for contribution!

**Pull request closed without merging**